### PR TITLE
Remove robust_ros3_retry

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
Removed the ROS3 retry helper since it is no longer needed

### CODE REVIEW

File: src/nwbinspector/_nwb_inspection.py [hunk 1]
```diff
@@ -145,8 +144,17 @@ def inspect_all(
...
         with pynwb.NWBHDF5IO(path=nwbfile_path, mode="r", load_namespaces=True) as io:
-            nwbfile = robust_s3_read(io.read)
-            identifiers[nwbfile.identifier].append(nwbfile_path)
+            try:
+                nwbfile = io.read()
+                identifiers[nwbfile.identifier].append(nwbfile_path)
+            except Exception as exception:
+                yield InspectorMessage(
+                    message=traceback.format_exc(),
+                    importance=Importance.ERROR,
```
Comment (CodyCBakerPhD) [inline src/nwbinspector/_nwb_inspection.py:153]: Also adding this in here now - it's a known problem that this pre-check can fail to collect and report messages if is read error occurs

Very hard to add tests for ATM though, might try in a follow-up that improves testing all around

File: src/nwbinspector/_nwb_inspection.py [hunk 2]
```diff
@@ -458,22 +466,30 @@ def run_checks(
...
-                        output.importance = check_function.importance
-                    yield output
-                elif output is not None:
-                    for x in output:
-                        x.importance = check_function.importance
-                        yield x
+            if check_function.neurodata_type is not None and not issubclass(
+                type(nwbfile_object), check_function.neurodata_type
+            ):
+                continue
```
Comment (CodyCBakerPhD) [inline src/nwbinspector/_nwb_inspection.py:440]: Broke up this logic to reduce cyclomatic complexity

File: tests/unit_tests/test_time_series.py [hunk 3]
```diff
@@ -371,12 +372,10 @@ def test_check_none_matnwb_resolution_pass():
...
-        nwbfile = robust_s3_read(command=io.read)
-        time_series = robust_s3_read(
-            command=nwbfile.processing["video_files"]["video"].time_series.get,
-            command_args=["20170203_KIB_01_s1.1.h264"],
-        )
-    assert check_resolution(time_series) is None
+        nwbfile = io.read()
+        time_series = nwbfile.processing["video_files"]["video"]["20170203_KIB_01_s1.1.h264"]
+
+        assert check_resolution(time_series=time_series) is None
```
Comment (CodyCBakerPhD) [inline tests/unit_tests/test_time_series.py:378]: This one will just be removed in a follow-up. Not too concerned about it now; has to do with old versions of packages

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,10 @@
 # Upcoming
 
+### Improvements
+* Removed the `robust_ros3_retry` utility helper. [#506](https://github.com/NeurodataWithoutBorders/nwbinspector/issues/506)
```
Comment (stephprince) [inline CHANGELOG.md]:
```suggestion
* Removed the `robust_ros3_read` utility helper. [#506](https://github.com/NeurodataWithoutBorders/nwbinspector/pull/506)
```

File: src/nwbinspector/_nwb_inspection.py (hunk 1 above)
Comment (stephprince) [reply src/nwbinspector/_nwb_inspection.py:153]: Sounds good. To clarify, at this point streaming is being redirected to `inspect_dandiset` earlier in `inspect_all`, so this except here is to catch local read errors?

**Review by stephprince — COMMENTED** [review]
Looks good to me!

File: src/nwbinspector/_nwb_inspection.py (hunk 1 above)
Comment (CodyCBakerPhD) [reply src/nwbinspector/_nwb_inspection.py:153]: Well, the check itself is a special one looking for duplicated file identifiers. That can't be done in an easy way as a normal check function, so it is done here before main deployment of the check registry

And yes, only applies to local folder contents. Have not implemented a Dandiset way of doing it yet

Comment (stephprince) [reply src/nwbinspector/_nwb_inspection.py:153]: Ah ok got it, thanks for explaining!

**Review by stephprince — APPROVED** [review]

**Pull request merged**